src/System.Management.Automation/FormatAndOutput/common/BaseOutputtingCommand.cs

```diff
@@ -1006,11 +1006,11 @@ internal override void ProcessPayload(FormatEntryData fed)
 
                 // need to make sure we have matching counts: the header count will have to prevail
                 string[] values = new string[headerColumns];
-                int[] alignment = new int[headerColumns];
+                Span<int> alignment = stackalloc int[headerColumns];
 
                 int fieldCount = tre.formatPropertyFieldList.Count;
 
-                for (int k = 0; k < headerColumns; k++)
+                for (int k = 0; k < values.Length; k++)
```

Comment:
Is there a reason for this change?  Seems like `headerColumns` is more readable

Reply:
In samples I always see `values.Length` - I guess the pattern Roslyn can recognize and optimize to remove bounder checks.

Reply:
Ok.

Reply:
I would prefer to use `headerColumns` here for better readability.

JIT is now able to detect and eliminate bounds check in the following scenario (see [this article](https://blogs.msdn.microsoft.com/dotnet/2017/10/16/ryujit-just-in-time-compiler-optimization-enhancements/)):
```
    int length = data.Length;
    byte x = 0;

    for (var idx = 0; idx < length; idx++)
    {
        x ^= data[idx];  // <---- bounds check here removed
    }
```
So I guess JIT is possibly able to detect the following too:
```
   string[] values = new string[headerColumns];
   int[] alignment = new int[headerColumns];
   for (int k = 0; k < headerColumns; k++)
   {
      values[k] = ...
      alignment[k] = ...
   }
```

Even JIT cannot do that yet, I'm not too concerned about the bounds check here, for two reasons:
1. We cannot eliminate all bounds check.
2. The bottleneck of formatting code is reflection, so micro-optimization like this should come after the readability.

Reply:
I reverted the change.

But I believe the patter is not so safe as `values.Lenght` because we could change to `string[] values = new string[headerColumns - 1];` and catch an error.